src/main/java/policycompass/fcmmanager/controllers/FCMModels.java

```diff
@@ -221,9 +226,26 @@ public static FCMModelDetail updateFCMModel(int id, JSONObject jsonModel) {
 		Session session = HibernateUtil.getSessionFactory().openSession();
 
 		session.beginTransaction();
-		Query qModel = session.createQuery("from fcmmanager_models where id= :id");
+		Query qModel;
+		if(adminUserFlag==1)
+			qModel = session.createQuery("from fcmmanager_models where id= :id");
+		else
+			qModel = session.createQuery("from fcmmanager_models where id= :id and userPath=:userPath");
```

Comment:
Don't do this. Simply retrieve the model and then validate the userPath. If the userPath doesn't match, check whether user is god.